crates/solver/src/solver/single_order_solver.rs

```diff
@@ -308,14 +346,29 @@ impl Solver for SingleOrderSolver {
         )
         .await;
 
+        let mut settlements: Vec<_> =
+            futures::future::join_all(settlements.into_iter().enumerate().map(
+                |(index, intermediate)| self.finalize_settlement(intermediate, &auction, index),
+            ))
+            .await
```

Comment:
Before this code, we ensure that creating the single order settlements doesn't take too long by making it part of the timeout. This code isn't part of the timeout but it can take a long time because it uses the network. That can make the solver take too long. I would combine the finalizing into the loop above. Can probably be expressed with `Stream`s where we have a pipeline order -> single order settlement -> finalize . When the timeout is hit or the order stream ends  we merge.